src/System.Private.CoreLib/shared/System/Buffers/ConfigurableArrayPool.cs

```diff
@@ -250,9 +252,12 @@ internal void Return(T[] array)
                 {
                     _lock.Enter(ref lockTaken);
 
-                    if (_index != 0)
+                    int index = _index;
+                    if (index != 0)
                     {
-                        _buffers[--_index] = array;
+                        index--;
+                        _buffers[index] = array;
+                        _index--;
```

Comment:
```suggestion
                        _index = index;
```
why not just assign the already decremented index?

Reply:
I'll have to check without tiered jit as @Suchiman suggested, but see my answer at https://github.com/dotnet/corefx/pull/41829#discussion_r335573397